All changes as of 22.08.25

Relican opened the pull request.

Relican committed:
All changes as of 22.08.25

KatyaProkhorchuk commented on manager_test.py:
```diff
@@ -0,0 +1,55 @@
+class TestCLI(unittest.TestCase):
+
+    def test_cli_valid_expressions(self):
+        os.mkdir(os.getcwd() + r'\TestCatalog')
```
путь нужно формировать через os.path.join или аналоги
этот вариант кода запуститься только на windows(
в windows в путях используется \ в то время как у macos/linux используется /

KatyaProkhorchuk commented on utils/funcs.py:
```diff
@@ -0,0 +1,137 @@
+        my_file.write(
+            "Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sun ...
+        my_file.close()
+        i += 1
```
зачем увеличивать i если это переменая цикла внутри for
и зачем в цикле кучу раз открывать файл заново? если хотите много записей почему не сделать 
```
my_file.write(
            "Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum. \n"*random.randint(1, 10000))
```

KatyaProkhorchuk commented on utils/funcs_test.py:
```diff
@@ -0,0 +1,101 @@
+        os.mkdir(os.getcwd() + r"\TestCatalog")
+        my_file = open(os.getcwd() + r"\TestCatalog\File1.txt", "a+")
+        my_file.write(
+            "Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sun ...
```
давайте вынесем эту строку в константу в файл config.py
много где используется это

Relican committed:
Вынес lorem ipsum в константу. Добавил os.path.join во всех директориях для совместимости. Исправил readme.md

KatyaProkhorchuk commented on utils/config.py:
супер, единственное(незначительное) замечание это то что согласно PEP 8, имена констант должны быть написаны прописными буквами с подчеркиваниями между словами:
LOREM=...

KatyaProkhorchuk commented on utils/funcs_test.py:
```diff
@@ -0,0 +1,108 @@
+        my_file.write(lorem)
+        my_file.close()
+
+        analyse_result = analyse(os.getcwd() + r"\TestCatalog")
```
путь лучше формировать через os.path.join

KatyaProkhorchuk commented on utils/funcs_test.py:
тесты предполагают наличие тестовой папки TestCatalog и каких-то файлов в ней
используйте setUp и tearDown в тестах для создания всех тестовых папок/файлов и удаления их в конце

Relican replied:
У меня в каждом отдельном тесте создается и удаляется каталог

KatyaProkhorchuk commented on utils/funcs_test.py:
```diff
@@ -0,0 +1,108 @@
+
+class MyTestCase(unittest.TestCase):
+    def test_copy(self):
+        os.mkdir(os.path.join(os.getcwd(), r"TestCatalog"))
```
os.mkdir() выбрасывает исключение FileExistsError, если пытаться создать директорию, которая уже существует.

Relican replied:
Добавил проверку на наличие пути перед созданием тестового каталога

KatyaProkhorchuk reviewed (changes requested):
пока тесты еще не работают(((
указала возможные причины

Relican committed:
Изменил имя константы. Добавил проверку на существующий тестовый каталог перед созданием.

KatyaProkhorchuk commented on utils/funcs_test.py:
во всех тестах повторяются строки с созданием директории и удалением
лучше использовать например так
```
def setUp(self):
        """Создание временной тестовой директории перед каждым тестом"""
        self.test_dir = tempfile.mkdtemp()
        self.test_file = os.path.join(self.test_dir, "File1.txt")
        
        # Создаем тестовый файл с содержимым
        with open(self.test_file, "w", encoding="utf-8") as f:
            f.write(LOREM)
    
    def tearDown(self):
        """Удаление временной директории после каждого теста"""
        shutil.rmtree(self.test_dir)
    
```

KatyaProkhorchuk replied:
так же os.rmdir в случае удаления непустой директории кидает исключение
нужно либо добавить проверку, либо использовать shutil.rmtree(self.test_dir)

KatyaProkhorchuk commented on utils/funcs_test.py:
```diff
@@ -0,0 +1,122 @@
+        my_file.write(LOREM)
+        my_file.close()
+
+        self.assertTrue(get_folder_size(os.path.join(os.getcwd(), r"TestCatalog")) == 448)
```
лучше сравнивать не с конкретным значением а например так:
 self.assertGreater(folder_size, 0)

все таки это размер папки и в каждой ос может быть разный, у меня к примеру тест упал(

Relican replied:
Не подумал о том, что размер будет разный :)

KatyaProkhorchuk commented on utils/funcs.py:
```diff
@@ -0,0 +1,118 @@
+
+
+# кол-во файлов в папке
+def nf(path):
```
не понятное название функции

KatyaProkhorchuk commented on utils/funcs.py:
```diff
@@ -0,0 +1,118 @@
+# создание файлового каталога
+def creation():
+    if os.path.exists(os.path.join(os.getcwd(), r"FileCatalog")):
+        os.rmdir(os.path.join(os.getcwd(), r"FileCatalog"))
```
как уже писала в тестах использование os.rmdir может кинуть исключение при удалении непустой директории

KatyaProkhorchuk commented on utils/funcs.py:
```diff
@@ -0,0 +1,118 @@
+    os.makedirs(os.path.join(os.getcwd(), r"FileCatalog", r"Catalog4", r"Catalog4_2"))
+    os.makedirs(os.path.join(os.getcwd(), r"FileCatalog", r"Catalog5", r"Catalog5_1"))
+
+    my_file = open(os.path.join(os.getcwd(), r"FileCatalog", r"Catalog1", r"File1.txt"), "a+")
+    my_file.write(LOREM * random.randint(1, 10000))
+    my_file.close()
```
лучше 
```
with open(os.path.join(os.getcwd(), r"FileCatalog", r"Catalog1", r"File1.txt"), "a+") as file:
   file.write(LOREM * random.randint(1, 10000))

```
тогда не надо думать о закрытии файла

KatyaProkhorchuk replied:
аналогичное замечание к тому что ниже

Relican committed:
Добавил в тесты setUp и tearDown.
Исправил функцию удаления тестового каталога.
Исправил создание тестового файла.

KatyaProkhorchuk commented on utils/funcs.py:
```diff
@@ -0,0 +1,110 @@
+        if os.path.isdir(folder_path):
+            folder_size = get_folder_size(folder_path)
+            folder_list.append(f"Папка: {folder_name} — Размер: {folder_size / (1024 * 1024):.2f} MB")
+    return "\n".join(folder_list)
```
метод считает только для папок без учета файлов

KatyaProkhorchuk reviewed (commented):
теперь все хорошо, за исключением неточностей в analyse

Relican committed:
Исправил функцию analyse

Relican committed:
Исправил тест для analyse
